koiki/delivery.py

```diff
@@ -1,25 +1,54 @@
 import base64
+import os
 
+# TODO: convert this class into Django Model in order to link Koiki's shipping, with WC purchase and PDF
+from django.core.mail import EmailMessage
+from django.template.loader import render_to_string
+from koiki import wcfmmp_host, logger
 
-class Delivery():
 
-    filename = 'label.pdf'
+class Delivery:
+    pdf_folder = "pdf_barcodes"
 
-    def __init__(self, data):
+    def __init__(self, data, vendor):
         self.data = data
+        self.vendor = vendor
+        self.shipment_id = data["numPedido"]
+        self.order_id = data["order_id"]
 
     def to_dict(self):
         return {
-            'number': self.data['numPedido'],
-            'barcode': self.data['codBarras'],
-            'label': self.data['etiqueta']
+            "number": self.data["numPedido"],
+            "barcode": self.data["codBarras"],
+            "label": self.data["etiqueta"],
         }
 
+    def _is_errored(self):
+        return self.data.get("respuesta", "") != "101"
+
     def print_pdf(self):
-        content = base64.b64decode(self.data['etiqueta'])
+        content = base64.b64decode(self.data["etiqueta"])
 
-        pdf = open(self.filename, 'wb')
+        pdf = open(os.path.join(self.pdf_folder, f"{self.shipment_id}.pdf"), "wb")
         pdf.write(content)
         pdf.close()
 
         return pdf
+
+    def send_mail_to_vendor(self):
+        self.print_pdf()
+        context = {
+            'wcfmmp_host': wcfmmp_host,
+            'wc_order_id': self.order_id,
+        }
+        logger.info('Sending Koiki pdf to email {}'.format(self.vendor.email))
+        message = render_to_string('contact_template.txt', context)
+        send_mail = EmailMessage(
+            f"Enviament Koiki per a la comanda: {self.order_id}",
+            message,
+            to=[self.vendor.email]
+        )
+        pdf = open(os.path.join(self.pdf_folder, f"{self.shipment_id}.pdf"), "rb")
+        send_mail.attach(f"{self.shipment_id}.pdf", pdf.read(), "application/pdf")
+
+        send_mail.send(fail_silently=False)
```

Comment:
It's fine for now, I wasn't sure when we added `print_pdf` but with `send_mail_to_vendor` I'm pretty sure it shouldn't live here. 

This class got now a few different reasons to change and they are not related, which makes it clear that it has many responsibilities. As a result, while in a separate class you may have extracted private helper methods out of `send_mail_to_vendor`, the fact this an extra responsibility that doesn't fit existing ones prevents us from doing the same here.

We can extract this into a `VendorOrderEmail` class perhaps or more imperative `SendMailToVendor`. That last option is a common pattern in Rails apps, see _2. Extract Service Objects_ in https://codeclimate.com/blog/7-ways-to-decompose-fat-activerecord-models/.

Reply:
It will change when we refactor in the next PR's to convert part of that logic into Django model.

Reply:
can we create an issue here as well?